src/main/java/com/fasterxml/jackson/databind/deser/std/ObjectArrayDeserializer.java

```diff
@@ -385,6 +385,11 @@ protected Object[] handleNonArray(JsonParser p, DeserializationContext ctxt)
 
         if (_untyped) {
             result = new Object[1];
+        } else if (value != null && !value.getClass().isAssignableFrom(_elementClass)) {
```

Comment:
I think the isAssignableFrom check should be reversed, if I’m reading this correctly. If the element type is Object and the value is a string, assigning a string to an element of an object array is allowed (string is assignable to object). The inverse isn’t the case, though.

Reply:
reversing the order of the classes in the assignableFrom check breaks 4 tests

```
[ERROR] Errors: 
[ERROR]   EmptyStringAsSingleValueTest.testBlankToArrayWrapper:163 » JsonMapping Unexpected IOException (of type java.io.IOException): The deserialized value class com.fasterxml.jackson.databind.convert.EmptyStringAsSingleValueTest$StringWrapper cannot be added to an array for classes of type com.fasterxml.jackson.databind.convert.EmptyStringAsSingleValueTest$StringWrapper
[ERROR]   EmptyStringAsSingleValueTest.testEmptyToArrayWrapper:127 » JsonMapping Unexpected IOException (of type java.io.IOException): The deserialized value class com.fasterxml.jackson.databind.convert.EmptyStringAsSingleValueTest$StringWrapper cannot be added to an array for classes of type com.fasterxml.jackson.databind.convert.EmptyStringAsSingleValueTest$StringWrapper
[ERROR]   CollectionDeserTest.testImplicitArrays:199 » JsonMapping Unexpected IOException (of type java.io.IOException): The deserialized value class com.fasterxml.jackson.databind.deser.jdk.CollectionDeserTest$XBean cannot be added to an array for classes of type com.fasterxml.jackson.databind.deser.jdk.CollectionDeserTest$XBean
[ERROR]   FormatFeatureAcceptSingleTest.testSingleElementArrayRead:231 » JsonMapping Unexpected IOException (of type java.io.IOException): The deserialized value class com.fasterxml.jackson.databind.struct.FormatFeatureAcceptSingleTest$Role cannot be added to an array for classes of type com.fasterxml.jackson.databind.struct.FormatFeatureAcceptSingleTest$Role
[I
```

whereas my code breaks zero tests

I suspect that the code would be easier to maintain if I change it to catch the ArrayStoreException that happens in the fuzz case instead.

Reply:
I'd much prefer proper check for assignability. I think catching exceptions should be the last effort in case where nothing else works.

But more important than this, even if either approach works, we do need a unit test to verify the fix.

Reply:
closing - I have no idea how the fuzzing framework causes the issue

Reply:
Yeah unfortunately it is very hard to reason about those failures. :-(